src/smif/sample_project/models/water_supply.py

```diff
@@ -38,18 +38,25 @@ def simulate(self, data):
 
         current_interventions = data.get_current_interventions()
 
-        print("Current interventions: {}".format(current_interventions))
+        self.logger.debug("Current interventions: {}".format(current_interventions))
         number_of_treatment_plants = 2
 
         # Inputs
-        per_capita_water_demand = data.get_parameter('per_capita_water_demand').as_ndarray()  # liter/person
+        per_capita_water_demand = data.get_parameter(
+            'per_capita_water_demand').as_ndarray()  # liter/person
         population = data.get_data('population').as_ndarray()  # people
 
         water_demand = data.get_data('water_demand').as_ndarray()  # liter
         final_water_demand = (population * per_capita_water_demand) + water_demand
 
         raininess = sum(data.get_data('precipitation').as_ndarray())  # milliliters to mega
-        reservoir_level = sum(data.get_data('reservoir_level').as_ndarray())  # megaliters
+        if data.current_timestep == data.base_timestep:
```

Comment:
There is some code in the data_handle.get_data() method (`_resolve_dependency`) which seems to do something similar.  Is it worth removing that, and ensuring that users explicitly define which data source to use in base timesteps and future timesteps, or do we want it to be resolved automatically in the data_handle?

Reply:
The code in data_handle decides which source to pull from - scenario or previous model result - depending on whether we're in the base year. I think this is okay - it avoids having two differently-named inputs, one of which is only provided with data in the base year, the other only in non-base years.

I thought it was more ambiguous how best to handle a request for data from a timestep previous to the base timestep - always `base_timestep - 1`? `base_timestep - inferred_timestep_stride`? just `base_timestep`?

Reply:
Just logged this for now: https://www.pivotaltracker.com/story/show/162482327